master/buildbot/test/fake/httpclientservice.py

```diff
@@ -78,20 +79,23 @@ def updateHeaders(self, headers):
 
     @classmethod
     def getFakeService(cls, master, case, *args, **kwargs):
-        ret = cls.getService(master, *args, **kwargs)
+        warn_deprecated('2.9.0', 'getFakeService() has been deprecated, use getService()')
+        return cls.getService(master, case, *args, **kwargs)
 
+    @classmethod
+    @defer.inlineCallbacks
+    def getService(cls, master, case, *args, **kwargs):
```

Comment:
I think the difference is that getFakeService does not have the same signature, as it has the case argument.

Reply:
Yes. On the other hand, the caller will need to know the initialization arguments of the service class one wants to instantiate/retrieve, so the real signature of each `getService` is different anyway.